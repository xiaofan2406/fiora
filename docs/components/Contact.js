/* @flow */
import React from 'react';
import { css } from 'react-emotion';
import { Loader } from 'widgets';

const cssContact = css`
  text-align: center;
`;

const Contact = () => (
  <div className={cssContact}>
    <Loader size={120} />
  </div>
);

export default Contact;
