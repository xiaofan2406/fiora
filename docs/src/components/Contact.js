/* @flow */
import React from 'react';
import { Input, Loader } from 'widgets';
import { delay, signUp } from 'utils/api';
import {
  usernameValidation,
  passwordValidation,
  passwordRepeatValidation,
} from 'utils/validations';
import { createFiora } from 'fiora';

const { Form, Field, FormMeta } = createFiora();

const signUpSubmit = async (data: { username: string, password: string }) => {
  const res = await signUp(data);
  if (res.status === 200) {
    console.log('Sign up successful!');
    return null;
  }
  return res.errors;
};

const Contact = () => (
  <Form
    onSubmit={signUpSubmit}
    onValidate={() => delay(2000)}
    onReset={() => {
      console.log('reset');
    }}
    data-testid="signUpForm"
  >
    <FormMeta>
      {({ error, isValidating, isSubmitting }) => (
        <>
          <div>{error}</div>
          {isValidating ? <Loader color="orange" size={32} /> : null}
          {isSubmitting ? <Loader color="green" size={32} /> : null}
        </>
      )}
    </FormMeta>
    <Field name="username" onValidate={usernameValidation}>
      {({
        value,
        error,
        isTouched,
        isValidating,
        handleChange,
        handleValidate,
      }) => (
        <Input
          data-testid="usernameInput"
          label="Username"
          value={value}
          error={error}
          isTouched={isTouched}
          isValidating={isValidating}
          onChange={event => {
            handleChange(event.target.value);
            handleValidate();
          }}
        />
      )}
    </Field>
    <Field name="password" onValidate={passwordValidation}>
      {password => (
        <Field
          name="passwordRepeat"
          onValidate={passwordRepeatValidation(password.value)}
        >
          {passwordRepeat => (
            <>
              <Input
                data-testid="passwordInput"
                label="Password"
                value={password.value}
                error={password.error}
                isTouched={password.isTouched}
                isValidating={password.isValidating}
                onChange={event => {
                  password.handleChange(event.target.value);
                }}
                onBlur={password.handleValidate}
              />
              <Input
                data-testid="passwordRepeatInput"
                label="Confirm password"
                value={passwordRepeat.value}
                error={passwordRepeat.error}
                isTouched={passwordRepeat.isTouched}
                isValidating={passwordRepeat.isValidating}
                onChange={event => {
                  passwordRepeat.handleChange(event.target.value);
                }}
                onBlur={passwordRepeat.handleValidate}
              />
            </>
          )}
        </Field>
      )}
    </Field>

    <button type="reset">Reset</button>
    <button type="submit">Submit</button>
  </Form>
);

export default Contact;
