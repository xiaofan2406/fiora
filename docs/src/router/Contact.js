import React from 'react';
import { Loader } from 'widgets';

const Contact = () => <Loader />;

export default Contact;
