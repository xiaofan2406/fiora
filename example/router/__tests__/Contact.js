import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Contact from '../Contact';

it('should match snapshot', () => {
  const component = shallow(<Contact />);

  expect(toJson(component)).toMatchSnapshot();
});
