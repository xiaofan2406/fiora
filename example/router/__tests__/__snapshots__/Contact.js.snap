// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<Loader
  color="#e8e8e8"
  size={60}
/>
`;
